src/coreclr/jit/jitconfigvalues.h

```diff
@@ -586,6 +586,8 @@ OPT_CONFIG_INTEGER(JitDoIfConversion, "JitDoIfConversion", 1)
 OPT_CONFIG_INTEGER(JitDoOptimizeMaskConversions, "JitDoOptimizeMaskConversions", 1) // Perform optimization of mask
                                                                                     // conversions
 
+RELEASE_CONFIG_INTEGER(JitOptimizeAwait, "JitOptimizeAwait", 1) // Perform optimization of Await intrinsics
```

Comment:
I wouldn't add a release knob for this.

Reply:
I have added this knob for two reasons:
1. we claim the optimization is optional, but I found couple cases which would fail without it. I think the scenarios are reachable in actual code, but disabling the optimization allows to do extra "stress" for calling via thunks. 
2. I wanted to see the impact of the optimizations on benchmarks.
 
I think both needs are temporary and we will not need the knob in the long run.

I think `#2` requires that it is a release knob. Is that correct?  Otherwise managed runtime parts will have asserts.
I assumed that is the reason why a bunch of other optimization related knobs are release knobs.

In any case, I`d prefer to log a tracking issue to remove this knob eventually. It seems useful right now, but at some point there will be no need.

Reply:
Ok, let's have it for now but make sure we remove it before any form of release that would lock its existence in.

Reply:
https://github.com/dotnet/runtimelab/issues/3012